pkg/eks/api.go

```diff
@@ -187,6 +187,9 @@ func newAWSProvider(spec *api.ProviderConfig, configurationLoader AWSConfigurati
 	}
 
 	if spec.Region == "" {
+		if cfg.Region == "" {
+			return nil, fmt.Errorf("no region code: please set the AWS region to ~/.aws/config file or environment variables")
+		}
 		spec.Region = cfg.Region
```

Comment:
https://github.com/eksctl-io/eksctl/blob/4e3d0e45dd37db58c2718c130d6bb5b9f4f7b334/pkg/eks/apiv2.go#L48-L50

Looks like `cfg.Region` is set from `spec.Region` in `newV2Config()` on L184. 

We should just move this check on `spec.Region` to before L184 and error if it's empty then.

Reply:
Thank you for your suggestion.

I understood the current logic as the following;
 * if `Spec.Region` is not empty, `cfg.Region` is set from `spec.Region` in `newV2Config()`
 * if `Spec.Region` is empty, `cfg.Region` is set from `~/.aws/config` or `AWS_REGION` in `newV2Config()`

So, I think we should check the value of `cfg.Region` after L184.
If `cfg.Region` is empty, we can determine AWS region is not defined.

//note
```
	cfg, err := newV2Config(spec, credentialsCacheFilePath, configurationLoader)
	if err != nil {
		return nil, err
	}

	fmt.Printf("spec.Region : %v\n", spec.Region)
	fmt.Printf("cfg.Region : %v\n", cfg.Region)

	if cfg.Region == "" {
		return nil, fmt.Errorf("AWS Region must be set, please set the AWS Region in AWS config file or as AWS_REGION environment variable")
	}
```

[result]
```
$ ./eksctl create cluster --dry-run
spec.Region : 
cfg.Region : ap-northeast-1

$ rm ~/.aws/config
$ ./eksctl create cluster --dry-run 
spec.Region : 
cfg.Region : 
Error: AWS Region must be set, please set the AWS Region in AWS config file or as AWS_REGION environment variable

$ export AWS_REGION=us-east-1      
$ ./eksctl create cluster --dry-run
spec.Region : 
cfg.Region : us-east-1

$ ./eksctl create cluster -f sample.yaml --dry-run
spec.Region : ap-northeast-1
cfg.Region : ap-northeast-1
```